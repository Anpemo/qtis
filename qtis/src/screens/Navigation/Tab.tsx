import React from 'react'
import { TouchableOpacity, StyleSheet } from 'react-native'
import { AntDesign, FontAwesome5, MaterialCommunityIcons } from '@expo/vector-icons'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 5,
    height: 50
  }
})
function Tab ({ color, onPress, icon }: any) {
  return (
        <TouchableOpacity style={styles.container} onPress={onPress} testID={'tabContainer'}>
            {icon && icon === 'search1' &&
            <AntDesign name="search1" size={30} color={color} testID={'search'} />
            }
            {icon && icon === 'barcode-scan' &&
              <MaterialCommunityIcons name="barcode-scan" size={30} color={color} testID={'barcodeScan'} />
            }
            {icon && icon === 'user' &&
              <FontAwesome5 name="user" size={30} color={color} testID={'user'}/>
            }
        </TouchableOpacity>
  )
}

export default Tab
