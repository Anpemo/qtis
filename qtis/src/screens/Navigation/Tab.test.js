import React from 'react'
import Tab from './Tab'
import { cleanup, render } from '@testing-library/react-native'

describe('Given a Tab component', () => {
  let component
  beforeEach(() => {
    component = (
              <Tab />
    )
  })
  afterEach(cleanup)
  describe('When calling it', () => {
    test('Then it will render a tabContainer', () => {
      const tabComponent = render(component)
      expect(tabComponent).toMatchSnapshot()
    })
  })
})
