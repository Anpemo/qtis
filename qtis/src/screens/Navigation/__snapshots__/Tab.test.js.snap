// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Tab component When calling it Then it will render a tabContainer 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 5,
    }
  }
  testID="tabContainer"
/>
`;
